contracts/andromeda_mission/src/testing/mod.rs

```diff
@@ -0,0 +1,585 @@
+    };
+    let info = mock_info("creator", &[]);
+
+    // we can just call .unwrap() to assert this was a success
+    let res = instantiate(deps.as_mut(), mock_env(), info, msg).unwrap();
+    assert_eq!(0, res.messages.len());
+}
+
+#[test]
+fn test_instantiation() {
+    let mut deps = mock_dependencies_custom(&[]);
+
+    let msg = InstantiateMsg {
+        operators: vec![],
+        mission: vec![MissionComponent {
+            name: "token".to_string(),
+            ado_type: "cw721".to_string(),
+            instantiate_msg: to_binary(&true).unwrap(),
+        }],
+        xfer_ado_ownership: false,
+        name: String::from("Some Mission"),
+        primitive_contract: String::from("primitive_contract"),
+    };
+    let info = mock_info("creator", &[]);
+
+    let res = instantiate(deps.as_mut(), mock_env(), info, msg).unwrap();
+    assert_eq!(1, res.messages.len());
+    match res.messages[0].msg.clone() {
+        CosmosMsg::Wasm(WasmMsg::Instantiate { code_id, msg, .. }) => {
+            assert_eq!(4, code_id);
+            assert_eq!(to_binary(&true).unwrap(), msg)
+        }
+        _ => panic!("Invalid msg type"),
+    }
+}
+
+// TODO: Figure out how to test this
+// #[test]
+// fn test_instantiation_xfer_ownership() {
+//     let mut deps = mock_dependencies_custom(&[]);
+
+//     let msg = InstantiateMsg {
+//         operators: vec![],
+//         mission: vec![MissionComponent {
+//             name: "token".to_string(),
+//             ado_type: "cw721".to_string(),
+//             instantiate_msg: to_binary(&true).unwrap(),
+//         }],
+//         xfer_ado_ownership: true,
+//         name: String::from("Some Mission"),
+//         primitive_contract: String::from("primitive_contract"),
+//     };
+//     let info = mock_info("creator", &[]);
+
+//     ADO_ADDRESSES
+//         .save(
+//             deps.as_mut().storage,
+//             "token",
+//             &Addr::unchecked("tokenaddress"),
+//         )
+//         .unwrap();
+
+//     let res = instantiate(deps.as_mut(), mock_env(), info, msg).unwrap();
+//     assert_eq!(2, res.messages.len());
+
+//     match res.messages[0].msg.clone() {
+//         CosmosMsg::Wasm(WasmMsg::Instantiate { code_id, msg, .. }) => {
+//             assert_eq!(4, code_id);
+//             assert_eq!(to_binary(&true).unwrap(), msg)
+//         }
+//         _ => panic!("Invalid msg type"),
+//     }
+
+//     match res.messages[1].msg.clone() {
+//         CosmosMsg::Wasm(WasmMsg::Execute {
+//             contract_addr, msg, ..
+//         }) => {
+//             assert_eq!("tokenaddress".to_string(), contract_addr);
+//             assert_eq!(
+//                 to_binary(&AndromedaMsg::UpdateOwner {
+//                     address: "creator".to_string()
+//                 })
+//                 .unwrap(),
+//                 msg
+//             );
+//         }
+//         _ => panic!("Invalid msg type"),
+//     }
```

Comment:
Instead of doing this why not just construct the desired `Response` and check equality with `assert_eq`?